### DISCUSSION TITLE
The app isn't responding

### DISCUSSION THREAD

**Discussion by ccinsz** [beeware/briefcase#1304] [Q&A]
There is a problem, the app isn't responding.
![image](https://github.com/beeware/briefcase/assets/134466342/bae0cf9e-1c2f-449c-b711-0615e1acc282)
When popup this dialog, it raises like this: 
Thread[6,tid=15205,WaitingInMainSignalCatcherLoop,Thread*=0xb400007788b12dd0,peer=0x12c401a0,"Signal Catcher"]: reacting to signal 3

How can I solve this problem?

**Comment by freakboy3742** [comment]
It is impossible for us to provide any advice without detailed reproduction instructions. We don't know what "this dialog" is, or anything else about the app you're currently writing.

**Reply by ccinsz** [reply]
the code of app.py as follows:

import time
import httpx
import toga
from toga.style import Pack
from toga.style.pack import COLUMN

class demo0529(toga.App):

    def startup(self):
        main_box = toga.Box(style=Pack(direction=COLUMN))
        main_box.add(toga.Button('Button', on_press=self.mytest))

        self.main_window = toga.MainWindow(title=self.formal_name)
        self.main_window.content = main_box
        self.main_window.show()

    def mytest(self, button):
        for i in range(10):
            with httpx.Client() as client:
                client.post("http://127.0.0.1:7912/session/com.android.settings")
            time.sleep(2)


def main():
    return demo0529()

**Reply by freakboy3742** [reply]
The app is reporting that it isn't responding because you're blocking the app's event loop.

This issue, and the fix, is discussed in [step 8 of the BeeWare tutorial](https://docs.beeware.org/en/latest/tutorial/tutorial-8.html).

**Reply by ccinsz** [reply]
This problem is solved. Thanks for your help.
